src/core/task/return.cc

```diff
@@ -233,15 +235,25 @@ void ReturnValues::legion_serialize(void* buffer) const
     ptr                               = ptr + sizeof(uint32_t);
   }
 
-  for (auto ret : return_values_) {
-    uint32_t size = ret.size();
 #ifdef LEGATE_USE_CUDA
-    if (ret.is_device_value())
-      CHECK_CUDA(cudaMemcpyAsync(ptr, ret.ptr(), size, cudaMemcpyDeviceToHost, stream));
-    else
+  if (Processor::get_executing_processor().kind() == Processor::Kind::TOC_PROC) {
+    auto stream = cuda::StreamPool::get_stream_pool().get_stream();
+    for (auto ret : return_values_) {
+      uint32_t size = ret.size();
+      if (ret.is_device_value())
+        CHECK_CUDA(cudaMemcpyAsync(ptr, ret.ptr(), size, cudaMemcpyDeviceToHost, stream));
+      else
+        memcpy(ptr, ret.ptr(), size);
+      ptr += size;
+    }
+  }
 #endif
+  else {
```

Comment:
I think this dangling `else` will raise an error for you if `LEGATE_USE_CUDA` is not defined. Did you mean the macro else?

Reply:
https://github.com/nv-legate/legate.core/pull/503/commits/8484263a0c3458291d701611805aa39b625a8686